controllers/rhmi/rhmi_controller.go

```diff
@@ -1006,10 +1003,10 @@ func (r *RHMIReconciler) preflightChecks(installation *rhmiv1alpha1.RHMI, instal
 			err = r.Status().Update(context.TODO(), installation)
 			if err != nil {
 				log.Infof("error updating status", l.Fields{"error": err.Error()})
-				return ctrl.Result{}, err
+				return err
 			}
 
-			return ctrl.Result{}, err
+			return nil
```

Comment:
So if secret is missing, we are going to continue with reconcile and by-pass the preflights. In here I would suggest to create your own error and return it instead.